aws/src/integration/java/org/apache/iceberg/aws/glue/GlueCatalogNamespaceTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.glue;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.NamespaceNotEmptyException;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.junit.Assert;
+import org.junit.Test;
+import software.amazon.awssdk.services.glue.model.Database;
+import software.amazon.awssdk.services.glue.model.EntityNotFoundException;
+import software.amazon.awssdk.services.glue.model.GetDatabaseRequest;
+
+public class GlueCatalogNamespaceTest extends GlueTestBase {
+
+  @Test
+  public void testCreateNamespace() {
+    String namespace = getRandomName();
+    namespaces.add(namespace);
+    glueCatalog.createNamespace(Namespace.of(namespace));
+    Database database = glue.getDatabase(GetDatabaseRequest.builder().name(namespace).build()).database();
+    Assert.assertEquals(namespace, database.name());
+  }
+
+  @Test
+  public void testCreateDuplicate() {
+    String namespace = createNamespace();
+    AssertHelpers.assertThrows("should not create namespace with the same name",
+        AlreadyExistsException.class,
+        () -> glueCatalog.createNamespace(Namespace.of(namespace)));
+  }
+
+  @Test
+  public void testCreateBadName() {
+    AssertHelpers.assertThrows("should not create namespace with bad name",
+        NoSuchNamespaceException.class,
+        () -> glueCatalog.createNamespace(Namespace.of("db-1")));
```

Comment:
I don't think this should throw `NoSuchNamespaceException`. What about `ValidationException`?

Also, tests should generally check the error message as well.

Reply:
Should this also test a namespace with more than 1 part?

Reply:
Added. More details are directly tested in unit test of `IcebergToGlueConverter`

Reply:
Yes `ValidationException` sounds better for the situations, I will update.